src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,45 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_null_properties_in_path(model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    document = {"properties": model}
+    for path in paths:
+        try:
+            prop, resolved_path, parent = traverse(document, path)
+        except LookupError:
+            pass  # not found means nothing to delete
+        else:
+            if prop is None or prop == "":
+                key = resolved_path[-1]
+                del parent[key]
```

Comment:
I might be missing something but should this always be a value if no exception happened?
also it's removing keys from `parent` but we return a `document`.

Reply:
also I think that modifying values in place isn't a good habit even though we been practicing it. it implies that we are intentionally causing side effects.